sdk/digitaltwins/digitaltwins/src/digitalTwinsClient.ts

```diff
@@ -0,0 +1,922 @@
+  DigitalTwinsAddRelationshipResponse,
+  DigitalTwinsAddRelationshipOptionalParams,
+  DigitalTwinsUpdateRelationshipOptionalParams,
+  DigitalTwinsUpdateRelationshipResponse,
+  DigitalTwinsDeleteRelationshipOptionalParams,
+  DigitalTwinsSendTelemetryOptionalParams,
+  DigitalTwinsSendComponentTelemetryOptionalParams,
+  DigitalTwinsListRelationshipsResponse,
+  IncomingRelationship,
+  DigitalTwinsListIncomingRelationshipsResponse,
+  ModelData,
+  DigitalTwinModelsGetByIdResponse,
+  DigitalTwinModelsGetByIdOptionalParams,
+  DigitalTwinModelsAddResponse,
+  DigitalTwinModelsAddOptionalParams,
+  DigitalTwinModelsListResponse,
+  DigitalTwinModelsListOptionalParams,
+  DigitalTwinModelsListOptions,
+  EventRoutesGetByIdResponse,
+  EventRoute,
+  EventRoutesAddOptionalParams,
+  EventRoutesListNextResponse,
+  EventRoutesListOptionalParams,
+  QueryQueryTwinsResponse,
+  QuerySpecification
+} from "./generated/models";
+import { logger } from "./logger";
+
+export const SDK_VERSION: string = "1.0.0-preview.1";
+
+/**
+ * Client for Azure IoT DigitalTwins API.
+ */
+export class DigitalTwinsClient {
+  /**
+   * @internal
+   * @ignore
+   * A reference to the auto-generated AzureDigitalTwinsAPI
+   */
+  private readonly client: GeneratedClient;
+
+  /**
+   * Creates an instance of AzureDigitalTwinsAPI.
+   *
+   * Example usage:
+   * ```ts
+   * const { DigitalTwinsClient, ServiceClientCredentials } = require("@azure/digitaltwins");
+   *
+   * const client = new DigitalTwinsClient(
+   *   "<endpoint>",
+   *   new DefaultAzureCredential();
+   * );
+   * ```
+   * @param {string} endpoint The endpoint of the service. If options parameter is given than this value will be ignored.
+   * @param {TokenCredential} credential Used to authenticate requests to the service.
+   * @param {AzureDigitalTwinsAPI} [options] Used to configure the service client.
+   */
+  constructor(
+    endpoint: string,
+    credential: TokenCredential,
+    options: AzureDigitalTwinsAPIOptionalParams = {}
+  ) {
+    const authPolicy = bearerTokenAuthenticationPolicy(
+      credential,
+      "https://digitaltwins.azure.net/.default"
+    );
+
+    const { ...pipelineOptions } = options;
+
+    const libInfo = `azsdk-js-digitaltwins/${SDK_VERSION}`;
+    if (!pipelineOptions.endpoint) {
+      pipelineOptions.endpoint = endpoint;
+    }
+    if (!pipelineOptions.userAgentHeaderName) {
+      pipelineOptions.userAgentHeaderName = libInfo;
+    }
+    if (pipelineOptions.userAgent) {
+      pipelineOptions.userAgent = "";
+    }
+
+    const internalPipelineOptions: InternalPipelineOptions = {
+      ...pipelineOptions,
+      ...{
+        loggingOptions: {
+          logger: logger.info,
+          allowedHeaderNames: ["x-ms-request-id"]
+        }
+      }
+    };
+
+    const pipeline = createPipelineFromOptions(internalPipelineOptions, authPolicy);
+
+    this.client = new GeneratedClient(credential, {
+      ...{ endpoint },
```

Comment:
Otherwise doesn't work.

---
In reply to: [474958384](https://github.com/Azure/azure-sdk-for-js/pull/10560#discussion_r474958384) [](ancestors = 474958384)